src/endpoints/sources.js

```diff
@@ -400,100 +400,115 @@ async function sendAny(req, res, url, accept) {
  */
 function receiveSubresource(subresourceExtractor) {
   return async (req, res) => {
-    const method = "PUT";
     const subresource = subresourceExtractor(req);
 
     authz.assertAuthorized(req.user, authz.actions.Write, subresource.resource);
 
-    /* Proxy the data through us:
-     *
-     *    client (browser, CLI, etc) ⟷ us (nextstrain.org) ⟷ upstream source
-     */
-    // eslint-disable-next-line prefer-const
-    let headers = {
-      "Content-Type": subresource.mediaType,
-      ...copyHeaders(req, ["Content-Encoding", "Content-Length"]),
-
-      /* XXX TODO: Consider setting Cache-Control rather than relying on
-       * ambiguous defaults.  Potentially impacts:
-       *
-       *   - Our own fetch() caching, including in sendSubresource() above
-       *   - Our Charon endpoints, if upstream headers are sent to the browser?
-       *   - CloudFront caching (not sure about its default behaviour)
-       *   - Browsers, if fetched directly, such as by redirection
-       *
-       * I think a cautious initial value would be to set "private" or "public"
-       * depending on the Source and then always set "must-revalidate,
-       * proxy-revalidate, max-age=0".  This would allow caches (ours,
-       * browsers, CloudFront?) to store the data but always check upstream
-       * with conditional requests.
-       *   -trs, 7 Dec 2021
-       */
-    };
+    const urlCallback = async (method, headers) => await subresource.url(method, headers);
 
-    // Body of the request as a Node stream
-    let body = req;
+    return receiveAny(req, res,
+      urlCallback,
```

Comment:
_Minor, non-blocking_

The need for `urlCallback` here and it's relationship to `receiveAny` had me briefly scratching my head first. I think it would read clearer (at least to me! ;-) to inline the function into the argument list:

```js
return receiveAny(req, res,
  async (method, headers) => await subresource.url(method, headers),
  subresource.mediaType
);
```

or

```js
return receiveAny(req, res,
  subresource.url.bind(subresource),
  subresource.mediaType
);
```

Reply:
Inlined: f233f81f5cbd18de7ba3b95c2688dd0d89e26646.

I'm not familiar with `bind()`, gonna try to understand that one tomorrow.

Reply:
Now I understand how `bind()` works here, but choosing to keep the first option. I think it's more readable mostly because the parser on my text editor does not provide any context on `subresource.url`, so I have to manually search for [the `url` property under `class Subresource`](https://github.com/nextstrain/nextstrain.org/blob/64a338e92f9a7e3d413503455d523578f75cef60/src/sources/models.js#L190) to understand that it is a function which takes `method` and `headers` (same for [`urlFor` under `class GroupSource`](https://github.com/nextstrain/nextstrain.org/blob/64a338e92f9a7e3d413503455d523578f75cef60/src/sources/groups.js#L72)). Having the explicit call makes it easier to tell what's going on.